examples/webview/webview/app.py

```diff
@@ -22,6 +26,16 @@ def on_bad_js_result(self, result, *, exception=None):
     def on_webview_load(self, widget, **kwargs):
         self.label.text = "www loaded!"
 
+    def on_navigation_starting(self, widget, url):
+        print(f"on_navigation_starting: {url}")
+        allow = True
+        if not url.startswith(self.allowed_base_url):
+            allow = False
+            message = f"Navigation not allowed to: {url}"
+            dialog = toga.InfoDialog("on_navigation_starting()", message)
+            asyncio.create_task(self.dialog(dialog))
```

Comment:
It might be easier to make on_navigation_starting an async callback, and then `await` the response.